src/core.ts

```diff
@@ -188,6 +189,26 @@ export function onTrigger(key: string): void {
   }
 }
 
+export function onCustomLog(kvps: { [key: string]: any }): void {
+  if (state === State.Activated) {
+    const event: IEventData = {
+      type: CustomEventName,
+      state: kvps,
+      time: getTimestamp(),
+    };
+    addEvent(event);
+  }
+}
+
+export function onSetPageInfo(pageId: string, userId: string): void {
+  // only allow setting pageId and userId if Clarity isn't currently running
+  if (state === State.Loaded || state === State.Unloaded) {
```

Comment:
Should we trap the "else" case or at least log something? A general note would be to add diagnostic logging but have it disabled by default.